include/zcbor_decode.h

```diff
@@ -74,17 +79,17 @@ bool zcbor_tstr_expect(zcbor_state_t *state, struct zcbor_string *result);
  * @param[in]    string  The value to expect. A pointer to the string.
  * @param[in]    len     The length of the string pointed to by @p string.
  */
-static inline bool zcbor_bstr_expect_ptr(zcbor_state_t *state, uint8_t *ptr, size_t len)
+static inline bool zcbor_bstr_expect_ptr(zcbor_state_t *state, char const *ptr, size_t len)
 {
-    struct zcbor_string zs = { .value = ptr, .len = len };
-    
-    return zcbor_bstr_expect(state, &zs);
+	struct zcbor_string zs = { .value = (const uint8_t *)ptr, .len = len };
```

Comment:
Why are we going from char to uint8? And what is the purpose of having `char const *` ? Isn't that default behavior? (that you cannot alter the actual pointer coming in)

Reply:
This function is often used on string literals, e.g. via `zcbor_bstr_expect_lit()`. String literals have the type `const char` in C++, so there was a compiler error when implicitly converting them to `const uint8_t`. To avoid users having to cast their string literals when using the API, I could have changed `zcbor_bstr_expect_lit()` to do the casting, but that is unsafe, because it is a macro, so the argument could be any type, which would then be silently cast. Instead I did this, so the argument is type-checked, then cast. Does that answer your question?